cohorts/count.py

```diff
@@ -21,48 +21,48 @@
-    sample_variants = cohort.load_variants(**kwargs)
-    def count_func(sample):
-        if sample in sample_variants:
-            return len(sample_variants[sample])
+    patient_variants = cohort.load_variants(**kwargs)
+    def count_func(patient_id):
+        if patient_id in patient_variants:
+            return len(patient_variants[patient_id])
         return np.nan
     return count(cohort, count_func, count_col="snv_count")
 
 def nonsynonymous_snv_count(cohort, **kwargs):
-    sample_nonsynonymous_effects = cohort.load_effects(only_nonsynonymous=True, **kwargs)
-    def count_func(sample):
-        if sample in sample_nonsynonymous_effects:
-            return len(sample_nonsynonymous_effects[sample])
+    patient_nonsynonymous_effects = cohort.load_effects(only_nonsynonymous=True, **kwargs)
+    def count_func(patient_id):
+        if patient_id in patient_nonsynonymous_effects:
+            return len(patient_nonsynonymous_effects[patient_id])
         return np.nan
     return count(cohort, count_func, count_col="nonsynonymous_snv_count")
 
 def missense_snv_count(cohort, **kwargs):
-    sample_nonsynonymous_effects = cohort.load_effects(only_nonsynonymous=True, **kwargs)
-    sample_missense_effects = dict(
-        [(sample,
+    patient_nonsynonymous_effects = cohort.load_effects(only_nonsynonymous=True, **kwargs)
+    patient_missense_effects = dict(
+        [(patient_id,
           EffectCollection(
               [effect for effect in effects if type(effect) == Substitution]))
-         for (sample, effects) in sample_nonsynonymous_effects.items()])
-    def count_func(sample):
-        if sample in sample_missense_effects:
-            return len(sample_missense_effects[sample])
+         for (patient_id, effects) in patient_nonsynonymous_effects.items()])
+    def count_func(patient_id):
+        if patient_id in patient_missense_effects:
+            return len(patient_missense_effects[patient_id])
         return np.nan
     return count(cohort, count_func, count_col="missense_snv_count")
 
 def neoantigen_count(cohort, **kwargs):
-    sample_neoantigens = cohort.load_neoantigens(**kwargs)
-    def count_func(sample):
-        if sample in sample_neoantigens["sample_id"].unique():
-            return len(sample_neoantigens[sample_neoantigens["sample_id"] == sample])
+    patient_neoantigens = cohort.load_neoantigens(**kwargs)
+    def count_func(patient_id):
+        if patient_id in patient_neoantigens["patient_id"].unique():
+            return len(patient_neoantigens[patient_neoantigens["patient_id"] == patient_id])
         return np.nan
     return count(cohort, count_func, count_col="neoantigen_count")
 
 def count(cohort, count_func, count_col):
-    df = cohort.clinical_dataframe.copy()
-    df[count_col] = df[cohort.clinical_dataframe_id_col].map(count_func)
+    df = cohort.as_dataframe()
+    df[count_col] = df["patient_id"].map(count_func)
     original_len = len(df)
     df = df[~df[count_col].isnull()]
```

Comment:
This is not a change here, but I might prefer getting the dataframe with the NA values. Most numpy/pandas/seaborn functions are good about handling them and you don't lose the other info on the patient.  Can be handled elsewhere.